### ISSUE TITLE
How to prevent from KL loss collapse

### ISSUE THREAD

**Issue by zyj008** [1Konny/Beta-VAE#4]
Hello!  I found your kl divergence curve is flat after about 100k iterations.  When I train other VAE tasks, I found that the kl divergence is easier to converge than reconstruction loss. The training loss usually suffer from kl divergence collapse. Kl converges to a number close to zero. So I want to know how to keep KL divergence curve flat with beta>1?

**Comment by 1Konny** [comment]
Hi.
you can manually make KL-divergence fixed to specific values by penalizing it if it diverges from expected values. In my case, for example, I followed the second term of the equation (8) in the paper, "Understanding disentangling in $\beta$-VAE". you can also find corresponding code here:
https://github.com/1Konny/Beta-VAE/blob/master/solver.py#L168
